src/Modules/Family/FamilyContainer.js

```diff
@@ -1,216 +1,54 @@
+  const onSubmit = (data) => {
+    console.log(data);
   };
-
-  const handleSubmitConfirm = () => {
-    let title = "Are you sure you want to proceed?";
-    confirm(title, handleSubmit);
-  };
-
-  const handleSubmit = () => {
-    let familyDetails = {
-      familyMemberData: {
-        first_name: familyData.nameData
-          ? familyData.nameData.nameData.first_name
-          : "",
-        last_name: familyData.nameData
-          ? familyData.nameData.nameData.last_name
-          : "",
-        middle_name: familyData.nameData
-          ? familyData.nameData.nameData.middle_name
-          : "",
-        hoh: familyData.nameData ? familyData.nameData.nameData.hoh : "",
-        dob: familyData.nameData ? familyData.nameData.nameData.dob : "",
-        suffix: familyData.nameData ? familyData.nameData.nameData.suffix : "",
-        phoneNumber: familyData.nameData
-          ? familyData.nameData.nameData.phoneNumber
-          : "",
-        phoneNumberCheckBOx: familyData.nameData
-          ? familyData.nameData.nameData.phoneNumberCheckBOx
-          : "",
-        email: familyData.nameData ? familyData.nameData.nameData.email : "",
-        communicationPreference: familyData.nameData
-          ? familyData.nameData.nameData.communicationPreference
-          : "",
-      },
-
-      HouseHoldData: {
-        address: familyData.addressData
-          ? familyData.addressData.addressData.streetAddress
-          : "",
-        apt_number: familyData.addressData
-          ? familyData.addressData.addressData.aptNo
-          : "",
-        zipcode: familyData.zipData ? familyData.zipData.zipData.zip : "",
-      },
-
-      passwordData: {
-        password: familyData.passwordData
-          ? familyData.passwordData.passwordData.password
-          : "",
-      },
-
-      memberCountData: {
-        countSenior: familyData.memberCountData
-          ? familyData.memberCountData.memberCountData.countSenior
-          : "",
-        countMiddle: familyData.memberCountData
-          ? familyData.memberCountData.memberCountData.countMiddle
-          : "",
-        countJunior: familyData.memberCountData
-          ? familyData.memberCountData.memberCountData.countMiddle
-          : "",
-      },
-
-      pickuptData: {
-        pickupInfo: familyData.pickupData
-          ? familyData.pickupData.pickupData.pickupInfo
-          : "",
-        pickupName: familyData.pickupData
-          ? familyData.pickupData.pickupData.pickupName
-          : "",
-        pickupNumberPlate: familyData.pickupData
-          ? familyData.pickupData.pickupData.pickupNumberPlate
-          : "",
-      },
-    };
-
-    if (familyDetails) {
-    } else {
-      showMessage("error", "Something went wrong");
-    }
-  };
-
-  const buildFamilyData = (childFamilyData) => {
-    let dataKey = Object.keys(childFamilyData)[0];
-    familyData[dataKey] = childFamilyData;
-  };
-
-  const formErrors = (errors) => {
-    formError = errors;
-  };
-
```

Comment:
All of the above is handled with `react-hook-form`. I believe we should lean on that instead of creating a custom framework.  Makes the code cleaner and [will do anything that we need](https://react-hook-form.com/).

Reply:
Hi Sean & David,

We are really sorry for the late response, we didn't get any notifications from the GitHub.

We went through the react hook form in the earlier stage of development and found some re-rendering issues with that and it's not yet resolved. The re-rendering may be affected in the performance. The re-rendering issue is reported in two months before.  if found any other issues like re-rendering in the later stage of development then the customization may be a bit of difficult. Due to this we created a custom hook for validation and used throughout the application. We are able to reproduce the same in the feat/registration branch. If you're planning to use the same(react-hook-form) in the application then please let us know. We will remove the custom hook that we created.

Reply:
Re rendering? I need more information than that. Do you have a github issue to point to?

Reply:
The reported issue is mentioned in the below link.
[](https://github.com/react-hook-form/react-hook-form/issues/912)
The status is mentioned like closed but they added to their todo list.
We have tested with our code by the following steps.
Step 1. Put a console log inside the family container.
Step 2. Click on the submit button or no phone available checkbox without entering any values in the forms.
Step 3. Change the values in any textbox field then we can see that the console.log value is displayed in the first keystroke or if we clear that text box values.

Reply:
I think the repositories in midohio is in public access, please make it as private otherwise public can access the git hub code. Just before I added a react-hook-form issue in the above message, it will automatically reference to there repo from there anybody can access the git repo. because it is in public mode.

Reply:
I understand the issue thanks. `watch` is used 2x if I remember correctly. Pretty disappointing, but is non blocking in my opinion.

I'm not married to react-hook-form. I can use a different solution if it adds value to overall app. However I'm very reluctant to switch to a custom solution because we will be passing this app along to other devs.

A custom solution creates domain knowledge which we want to avoid at all cost. 

The developers on this project are volunteering and there will be a lot of turnover. Your custom built solution was not documented, not feature rich, and had no testing. New devs will be able to see  react-hook-form, read the documentation, and submit issues like you did when something pops up. 

[The issue is in the backlog](https://github.com/react-hook-form/react-hook-form/issues/912), which I think is a great sign of the community working on it.

I can be talked into a different third party, but I think a custom built solution should be strongly discouraged.

Reply:
Thank you Sean for your detailed information, we will remove the custom hook that we created and we will replace the existing code with react hook form. If you're planning to change other than react hook form, please share with us.

Reply:
@seanlawrenz thank you for the great thinking on this topic. 
@Basiltj07 & our InApp friends, thank you all for your flexibility in the approach.

In general I would say this conversation points to a good architecture decision - Like Sean said; avoid domain knowledge at all costs. We are trying to keep the project accessible to newcomers and are interested in any suggestions or documentation that can advance that aim.